File: scala-pr/54/scala/code-quality-logging/SecureUserAuthenticator.scala
class SecureUserAuthenticator {
  def authenticate(username: String, password: String): Boolean = {
    // {fact rule=code-quality-logging@v1.0 defects=0}
    println(s"Authentication attempt for user: $username")
    
    if (username == "admin" && password == "secret123") {
      println(s"Authentication successful for user: $username")
      true
    } else {
      println(s"Authentication failed for user: $username")
      false
    }
    // {/fact}
  }
}

File: php-pr/51/php/code-quality-logging/SecureUserAuthenticator.php
<?php

class SecureUserAuthenticator
{
    public function authenticate($username, $password)
    {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        error_log("Authentication attempt for user: {$username}");
        
        if ($username === 'admin' && $password === 'secret123') {
            error_log("Authentication successful for user: {$username}");
            return true;
        }
        
        error_log("Authentication failed for user: {$username}");
        return false;
        // {/fact}
    }
}